### PR TITLE
Feat: added `ssl_supported_protocols` option

### PR DESCRIPTION
**Pull request by kares**
actual option implementation provided by mixin https://github.com/logstash-plugins/logstash-mixin-http_client/pull/40

### CODE REVIEW

File: docs/index.asciidoc [hunk 1]
```diff
@@ -378,6 +379,24 @@ See <<plugins-{type}s-{plugin}-retry_policy,Retry Policy>> for more information.
...
+[id="plugins-{type}s-{plugin}-ssl_supported_protocols"]
+===== `ssl_supported_protocols`
+
+* Value type is <<string,string>>
+* Allowed values are: `TLSv1.1`, `TLSv1.2`, 'TLSv1.3'
+* There is no default value for this setting
+
+List of allowed SSL/TLS versions to use when establishing a connection to the HTTP endpoint.
+The default depends on the JDK being used and with up-to-date Logstash is `['TLSv1.2', 'TLSv1.3']`.
+`TLSv1.1` is not considered secure and is only provided for legacy applications.
```
Comment (karenzone) [inline docs/index.asciidoc:391]:
```suggestion
* Value type is <<string,string>>
* Allowed values are: `TLSv1.1`, `TLSv1.2`, `TLSv1.3`
* Default depends on the JDK being used. With up-to-date Logstash, the default is `['TLSv1.2', 'TLSv1.3']`.
`TLSv1.1` is not considered secure and is only provided for legacy applications.

List of allowed SSL/TLS versions to use when establishing a connection to the HTTP endpoint.
```

File: spec/outputs/http_spec.rb [hunk 2]
```diff
@@ -520,22 +415,28 @@ def start_app_and_wait(app, opts = {})
...
-    })
+    TestApp.server_settings = @@default_server_settings.merge(webrick_config)
 
     TestApp.last_request = nil
 
-    @server = start_app_and_wait(TestApp)
+    @server = start_app_and_wait(TestApp, &server_setup)
+  end
+
+  let(:server_setup) { lambda { |_| } }
```
Comment (kaisecheng) [inline spec/outputs/http_spec.rb:430]: why this lambda is required?

Comment (kaisecheng) [inline spec/outputs/http_spec.rb:430]: why this lambda is required?

**Review by kaisecheng — APPROVED** [review]
The actual change looks good. It takes the new option from mixin http_client. Thanks for the effort refactoring the test 🏅 . I left one question to the test

File: docs/index.asciidoc (hunk 1 above)
Comment (kares) [reply docs/index.asciidoc:391]: nope that wasn't - it's a typo, thanks for noticing

File: spec/outputs/http_spec.rb (hunk 2 above)
Comment (kares) [reply spec/outputs/http_spec.rb:430]: actually isn't required - thought I could use the hook to force the `context. ssl_version=`
but turns out it's too late and needs a patch: https://github.com/logstash-plugins/logstash-output-http/pull/131/files#diff-89eebfcbc0f14b6d989517837ca1e94fce4e2ce9a03233641cd936f2b8d2ed94R26

Comment (kaisecheng) [reply spec/outputs/http_spec.rb:430]: got it. You can consider removing lambda and start_app_and_wait(TestApp, ~~&server_setup~~)

**Pull request merged**